src/datasource.js

```diff
@@ -194,6 +207,24 @@ export class GenericDatasource {
     };
   }
 
+  createLogsQueryForAnnotation(options) {
+    const queryText = this.templateSrv.replace(options.annotation.queryText, options.scopedVars, GenericDatasource.interpolateVariable);
+
+    return {
+      url: this.url + '/query',
+      method: 'POST',
+      headers: this.headers,
+      data: JSON.stringify({
+        token: this.apiKey,
+        queryType: "log",
+        filter: queryText,
+        startTime: options.range.from.valueOf(),
+        endTime: options.range.to.valueOf(),
+        maxCount: 5000
```

Comment:
Is there a situation where we would want this value to be lower / higher (I assume there is some kind of upper limit on the API side)?

Reply:
5000 is the upper limit of the API side, if we wanted to retrieve more values we could repeat the requests with the continuation token, but as I found out while testing Grafana can't even handle 5000 entries. In fact I should probably lower this, but I don't know what value would be reasonable, possibly 200? I'm unsure.